### PR TITLE
Add libunity

### PR DESCRIPTION
**Pull request by JakobDev**
libunity is complete outdated, so it's a nightmare to build. It is [needed by Electron](https://github.com/electron/electron/blob/fb88375ab4d2161dbf7e958a2a94c7c6d97dc84c/shell/browser/linux/unity_service.cc#L74) for [setProgressBar](https://www.electronjs.org/de/docs/latest/api/browser-window#winsetprogressbarprogress-options) and [setBadgeCount](https://www.electronjs.org/de/docs/latest/api/browser-window#winsetprogressbarprogress-options). You can use this [example code](https://www.electronjs.org/de/docs/latest/tutorial/progress-bar) to test it out.

The app need the `--socket=session-bus` permission for some reason. You actually just [need to emit a DBus signal](https://gist.github.com/lephuongbg/1830d1e03ac45faa11e1bbd7204eee5d), which should work without any permission.

A big problem is, that this API uses the desktop file name. You need to set `desktopName` field in the `package.json` to the correct name, otherwise it uses the name if the App, which is not the Flatpak ID. So this will cause problems when repacking existing binaries. I ping @refi64. Maybe he has a Idea what to do about that.

Packagers should also set the `StartupWMClass` field in the desktop entry, otherwise the desktop file can't be connected to the App.

Another problem is, that [Electron checks for some reason the current desktop](https://github.com/electron/electron/blob/fb88375ab4d2161dbf7e958a2a94c7c6d97dc84c/shell/browser/linux/unity_service.cc#L64). This is supported by more Desktops e.g. Mate as the ones Electron checks, so packages should overwrite the `XDG_CURRENT_DESKTOP` environment variable to make it work on every desktop.

This PR is heavily based on [this PR here](https://github.com/flathub/com.discordapp.Discord/pull/167) and the AUR packages of the used software.

For reference:
flathub/com.discordapp.Discord#228
Heroic-Games-Launcher/HeroicGamesLauncher#3478

### CODE REVIEW

**Comment by Erick555** [comment]
> The app need the --socket=session-bus permission for some reason.

Does it make sense then when flathub won't allow that?

**Comment by JakobDev** [comment]
> Does it make sense then when flathub won't allow that?

You need to access a exception, if you want to use this Feature.

**Comment by Erick555** [comment]
I don't think you'll be allowed for --socket=session-bus exception on flathub therefore this may be unusable.

**Comment by bbhtt** [comment]
Seems fine as long as it continues to build with newer runtimes and works with apps. If it fails to build at some point it has to be removed for obvious reasons.

But `socket=session-bus` just for notification badges makes no sense and won't be allowed on apps.

This does not make it completely useless just changes the default. (Or the specific permission needed should be investigated)

**Comment by bbhtt** [comment]
Seems fine as long as it continues to build with newer runtimes and works with apps.

But socket=session-bus just for notification badges makes no sense and won't be allowed on apps.

This does not make it completely useless just changes the default. (Or the specific permission needed should be investigated)

**Comment by JakobDev** [comment]
Turns out I'm dumb. While trying the Permissions out I occidentally used the System Bus instead of the Session Bus in Flatseal, so I couldn't get it to world. Looks like you just need `--talk=com.canonical.Unity`. Sorry for that.

> Seems fine as long as it continues to build with newer runtimes and works with apps.

I would be far better, if Electron would just use the DBus Interface directly instead of relying n an outdated lib, but as building Electron also requires building Chromium, I'm not the one going to make the PR.

**Comment by JakobDev** [comment]
@TingPing Can you merge this?

File: libunity.yml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    buildsystem: autotools
+    build-options:
+        cflags: "-Wno-error"
+        env:
+            HAVE_VALGRIND_FALSE: "#"
+            HAVE_VALGRIND_TRUE: ""
+    config-opts:
+      - --disable-static
+      - --disable-gtk-doc
+      - --enable-introspection=yes
```
Comment (TingPing) [inline libunity.yml:81]: Disable all introspection.

File: org.electronjs.Electron2.BaseApp.yml [hunk 2]
```diff
@@ -4,6 +4,8 @@ runtime: org.freedesktop.Platform
 sdk: org.freedesktop.Sdk
 runtime-version: '23.08'
 separate-locales: false
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.vala
```
Comment (TingPing) [inline org.electronjs.Electron2.BaseApp.yml:8]: You don't need vala, rip it out.

File: libunity.yml (hunk 1 above)
Comment (TingPing) [reply libunity.yml:81]: Oh... libunity is in Vala. But this disables vala anyway so I guess its not needed.

**Comment by TingPing** [comment]
I'll merge this but you do understand it *will* break every single runtime update. You're responsible for it now. It will just be removed if it blocks future versions.

File: libunity.yml (hunk 1 above)
Comment (TingPing) [reply libunity.yml:81]: I was wrong, it depended on the gir files but not the vapi files..?

Comment (JakobDev) [reply libunity.yml:81]: It doesn't build with introspection disabled, so we should enable it

**Comment by JakobDev** [comment]
> It will just be removed if it blocks future versions.

As this lib is optional and in a extra file, it can just be commented out until a solution is found

File: libunity.yml (hunk 1 above)
Comment (TingPing) [reply libunity.yml:81]: Yes sorry about that. Just not what I expected.

**Pull request merged**